orchagent/bufferorch.cpp

```diff
@@ -170,14 +165,58 @@ void BufferOrch::generateBufferPoolWatermarkCounterIdList(void)
         statList.pop_back();
     }
 
-    vector<FieldValueTuple> fvTuples;
-    fvTuples.emplace_back(BUFFER_POOL_COUNTER_ID_LIST, statList);
+    // Some platforms do not support buffer pool watermark clear operation on a particular pool
+    // Invoke the SAI clear_stats API per pool to query the capability from the API call return status
+    unsigned int noWmClrCapability = 0;
+    unsigned int bitMask = 1;
```

Comment:
Why use a bit mask? What is the length of m_buffer_type_maps[CFG_BUFFER_POOL_TABLE_NAME]?

Reply:
It depends on the configuration. In our use case, it will be 3.